lib/commands/connect/shared.js

```diff
@@ -182,6 +182,19 @@ exports.withConnection = function(context, heroku) {
   });
 }
 
+var withMapping = exports.withMapping = function(connection, object_name) {
+  return Q.Promise(function(resolve, reject) {
+    var found = false, object_name_lower = object_name.toLowerCase();
+    connection.mappings.forEach(function(mapping) {
+      if (mapping.object_name.toLowerCase().indexOf(object_name_lower) == 0) {
```

Comment:
Perhaps, in the long run. It would change the API of this a bit, because you wouldn't need to fetch the connection first, so I think I'll hold off for now, in favor of what'll likely be the more common use case to start out with.